cpp/oneapi/dal/algo/linear_regression/test/online_spmd.cpp

```diff
@@ -0,0 +1,126 @@
+/*******************************************************************************
+* Copyright contributors to the oneDAL project
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/linear_regression/test/fixture.hpp"
+#include "oneapi/dal/test/engine/tables.hpp"
+#include "oneapi/dal/test/engine/io.hpp"
+
+namespace oneapi::dal::linear_regression::test {
+
+namespace te = dal::test::engine;
+namespace la = te::linalg;
+namespace linear_regression = oneapi::dal::linear_regression;
+
+template <typename TestType>
+class lr_online_spmd_test : public lr_test<TestType, lr_online_spmd_test<TestType>> {
+public:
+    using base_t = lr_test<TestType, lr_online_spmd_test<TestType>>;
+    using float_t = typename base_t::float_t;
+    using input_t = typename base_t::train_input_t;
+    using partial_input_t = typename base_t::partial_input_t;
+    using partial_result_t = typename base_t::partial_result_t;
+    using result_t = typename base_t::train_result_t;
+
+    void set_rank_count(std::int64_t rank_count) {
+        n_rank = rank_count;
+    }
+
+    std::int64_t get_rank_count() {
+        return n_rank;
+    }
+
+    void generate_dimensions() {
+        this->t_count_ = GENERATE(307, 12999);
```

Comment:
it makes sense to little bit reduce the number of tests for online linreg, check that it takes locally not too much time

Reply:
this is already reduced set of tests, takes 3.9 seconds. do you think this is good enough or should i reduce further?

Reply:
Probably increase up to ~20seconds